### ISSUE TITLE
OAuth Alongside Form Login

### ISSUE THREAD

**Issue by MrGrinst** [spring-projects/spring-security-oauth#241]
I'm trying to configure my application so that part of the app is accessible through OAuth. I still want form login for main users, however.

I'm using the following XML config for Spring Security:

```
    <?xml version="1.0" encoding="UTF-8"?>
    <beans xmlns:sec="http://www.springframework.org/schema/security"
       xmlns="http://www.springframework.org/schema/beans"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xmlns:oauth="http://www.springframework.org/schema/security/oauth2"
       xsi:schemaLocation="http://www.springframework.org/schema/beans
     http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
     http://www.springframework.org/schema/security
     http://www.springframework.org/schema/security/spring-security-3.2.xsd
     http://www.springframework.org/schema/security/oauth2
     http://www.springframework.org/schema/security/spring-security-oauth2.xsd">

<sec:global-method-security pre-post-annotations="enabled">
    <sec:expression-handler ref="expressionHandler"/>
</sec:global-method-security>

<import resource="classpath:jpaContext.xml"/>

<sec:http pattern="/data/oauth/token/**"
      create-session="stateless"
      authentication-manager-ref="clientAuthenticationManager"
      entry-point-ref="oauthAuthenticationEntryPoint"
      xmlns="http://www.springframework.org/schema/security">
    <intercept-url pattern="/data/oauth/token/**" access="IS_AUTHENTICATED_FULLY" />
    <anonymous enabled="false" />
    <custom-filter ref="clientCredentialsTokenEndpointFilter" before="BASIC_AUTH_FILTER" />
    <access-denied-handler ref="oauthAccessDeniedHandler" />
</sec:http>

<sec:http pattern="/data/exporter/**"
      create-session="never"
      entry-point-ref="oauthAuthenticationEntryPoint"
      xmlns="http://www.springframework.org/schema/security">
    <anonymous enabled="false"/>
    <intercept-url pattern="/data/exporter/**" access="IS_AUTHENTICATED_FULLY" />
    <custom-filter ref="resourceServerFilter" before="PRE_AUTH_FILTER" />
    <access-denied-handler ref="oauthAccessDeniedHandler" />
</sec:http>

<sec:http use-expressions="true" entry-point-ref="authenticationEntryPoint">
    <sec:intercept-url pattern="/login.html" access="!isAuthenticated()" />
    <sec:intercept-url pattern="/index.html" access="isAuthenticated()" />
    <sec:intercept-url pattern="/data/**" access="isAuthenticated()"/>
    <sec:form-login login-page="/login.html" authentication-failure-handler-ref="authenticationFailureHandler" authentication-success-handler-ref="authenticationSuccessHandler"/>
    <sec:logout delete-cookies="JSESSIONID,sessionContext" invalidate-session="true" logout-success-url="/login.html"/>
    <sec:access-denied-handler ref="accessDeniedHandler"/>
</sec:http>

<oauth:authorization-server
        client-details-service-ref="clientDetailsService"
        token-services-ref="tokenServices"
        user-approval-handler-ref="userApprovalHandler">
    <oauth:authorization-code />
    <oauth:refresh-token />
    <oauth:client-credentials />
    <oauth:password authentication-manager-ref="userAuthenticationManager" />
</oauth:authorization-server>

<oauth:resource-server id="resourceServerFilter" resource-id="data" token-services-ref="tokenServices" />

<sec:authentication-manager alias="userAuthenticationManager">
    <sec:authentication-provider user-service-ref="userDetailsService">
        <sec:password-encoder ref="encoder" />
    </sec:authentication-provider>
</sec:authentication-manager>

<sec:authentication-manager id="clientAuthenticationManager">
    <sec:authentication-provider user-service-ref="clientDetailsUserService" />
</sec:authentication-manager>

<bean id="oauthAuthenticationEntryPoint" class="com.example.webservices.security.OAuthAuthenticationEntryPoint">
    <property name="realmName" value="data" />
</bean>

<bean id="accessDecisionManager" class="org.springframework.security.access.vote.UnanimousBased" xmlns="http://www.springframework.org/schema/beans">
    <constructor-arg>
        <list>
            <bean class="org.springframework.security.oauth2.provider.vote.ScopeVoter" />
            <bean class="org.springframework.security.access.vote.AuthenticatedVoter" />
            <bean class="org.springframework.security.web.access.expression.WebExpressionVoter"/>
        </list>
    </constructor-arg>
</bean>


<!--TOKEN STUFF-->
<bean id="tokenStore" class="org.springframework.security.oauth2.provider.token.store.JdbcTokenStore">
    <constructor-arg ref="jdbcTemplate"/>
</bean>

<bean id="jdbcTemplate" class="org.springframework.jdbc.datasource.DriverManagerDataSource">
    <property name="driverClassName" value="${env.driver}"/>
    <property name="url" value="${env.url}"/>
    <property name="username" value="${env.username}"/>
    <property name="password" value="${env.password}"/>
</bean>

<bean id="tokenServices" class="org.springframework.security.oauth2.provider.token.DefaultTokenServices">
    <property name="tokenStore" ref="tokenStore" />
    <property name="supportRefreshToken" value="true" />
    <property name="clientDetailsService" ref="clientDetailsService" />
    <property name="accessTokenValiditySeconds" value="604800" />
</bean>

<bean id="oauthRequestFactory" class="org.springframework.security.oauth2.provider.request.DefaultOAuth2RequestFactory">
    <constructor-arg name="clientDetailsService" ref="clientDetailsService" />
</bean>

<bean id="userApprovalHandler" class="org.springframework.security.oauth2.provider.approval.TokenStoreUserApprovalHandler">
    <property name="tokenStore" ref="tokenStore" />
    <property name="requestFactory" ref="oauthRequestFactory" />
</bean>


<!--CLIENT DETAILS STUFF-->
<oauth:client-details-service id="clientDetailsService">
    <oauth:client client-id="webapp"
                  authorized-grant-types="password,authorization_code,refresh_token"
                  authorities="ROLE_CLIENT"
                  scope="webapp" />
</oauth:client-details-service>

<bean id="clientDetailsUserService"
      class="org.springframework.security.oauth2.provider.client.ClientDetailsUserDetailsService">
    <constructor-arg ref="clientDetailsService" />
</bean>


<bean id="clientCredentialsTokenEndpointFilter"
      class="org.springframework.security.oauth2.provider.client.ClientCredentialsTokenEndpointFilter">
    <property name="authenticationManager" ref="clientAuthenticationManager" />
</bean>

<bean id="clientAuthenticationEntryPoint" class="org.springframework.security.oauth2.provider.error.OAuth2AuthenticationEntryPoint">
    <property name="realmName" value="data" />
</bean>

<bean id="oauthAccessDeniedHandler" class="org.springframework.security.oauth2.provider.error.OAuth2AccessDeniedHandler" />


<bean id="forwardingLoginUrlAuthenticationEntryPoint" class="org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint">
    <constructor-arg index="0" value="/login.html"/>
    <property name="useForward" value="true"/>
</bean>
<bean id="userDetailsService" class="com.example.webservices.security.userDetailService"/>
<bean id="encoder" class="org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder"/>
<bean id="authenticationSuccessHandler" class="com.example.webservices.security.authenticationSuccessHandler"/>
<bean id="authenticationFailureHandler" class="com.example.webservices.security.CustomAuthenticationFailureHandler"/>
<bean id="authenticationEntryPoint" class="com.example.webservices.security.SecuredDataEntryPoint">
    <constructor-arg name="loginUrl" value="/login.html"/>
</bean>
<bean id="accessDeniedHandler" class="com.example.webservices.security.CustomAccessDeniedHandler"/>
```

</beans>

For the most part this works great. It allows me to log in using either form login or OAuth. However, when (in Intellij) I deploy the application with an application context other than "/", it won't allow me to login via OAuth. For instance, in Intellij if I launch a new instance of the server with the application context "/app", I get an unauthorized error when I try to login.

**TL;DR OAuth is only working when the application context is "/".**
